Library/Formula/samba.rb

```diff
@@ -21,11 +21,43 @@ class Samba < Formula
 
   def install
     cd "source3" do
+      # This stops samba dumping .msg and .dat files directly into lib
+      # It can't be set with a configure switch - There isn't one that fine-grained.
+      inreplace "configure", "${MODULESDIR}", "#{share}/codepages"
+
       system "./configure", "--disable-debug",
                             "--prefix=#{prefix}",
                             "--with-configdir=#{prefix}/etc",
                             "--without-ldap",
                             "--without-krb5"
+
+      # https://bugzilla.samba.org/show_bug.cgi?id=11113
+      # The first 3 blocks require a *FULL PATH* inreplace, not just the dylib, otherwise
+      # the inreplace does both dylibs at once which isn't intended behaviour and
+      # breaks inreplacing and thus compile.
+      inreplace "Makefile" do |s|
+        s.gsub! "LIBTALLOC_SHARED_TARGET_SONAME=bin/libtalloc.dylib.2",
+                "LIBTALLOC_SHARED_TARGET_SONAME=bin/libtalloc.2.dylib"
+        s.gsub! "LIBTALLOC_SHARED_TARGET_FULLVER=bin/libtalloc.dylib.2.0.5",
+                "LIBTALLOC_SHARED_TARGET_FULLVER=bin/libtalloc.2.0.5.dylib"
+
+        s.gsub! "LIBTEVENT_SHARED_TARGET_SONAME=bin/libtevent.dylib.0",
+                "LIBTEVENT_SHARED_TARGET_SONAME=bin/libtevent.0.dylib"
+        s.gsub! "LIBTEVENT_SHARED_TARGET_FULLVER=bin/libtevent.dylib.0.9.11",
+                "LIBTEVENT_SHARED_TARGET_FULLVER=bin/libtevent.0.9.11.dylib"
+
+        s.gsub! "LIBTDB_SHARED_TARGET_SONAME=bin/libtdb.dylib.1",
+                "LIBTDB_SHARED_TARGET_SONAME=bin/libtdb.1.dylib"
+        s.gsub! "LIBTDB_SHARED_TARGET_FULLVER=bin/libtdb.dylib.1.2.9",
+                "LIBTDB_SHARED_TARGET_FULLVER=bin/libtdb.1.2.9.dylib"
+
+        s.gsub! "libwbclient.dylib.0", "libwbclient.0.dylib"
+        s.gsub! "libaddns.dylib.0", "libaddns.0.dylib"
+        s.gsub! "libnetapi.dylib.0", "libnetapi.0.dylib"
+        s.gsub! "libsmbclient.dylib.0", "libsmbclient.0.dylib"
+        s.gsub! "libsmbsharemodes.dylib.0", "libsmbsharemodes.0.dylib"
+      end
```

Comment:
Just if you fancy a future challenge: I reckon you could do all of these with 1-2 regexes. Happy to pull as-is though.